toy_robot.rb

```diff
@@ -10,12 +10,25 @@ def initialize
     @movements = []
   end
 
+
+
   def place(x,y,facing)
-    raise InvalidPositionError.new("Invalid Position, #{x}, #{y}, #{facing}") if !(x.between?(0,4)) || !(y.between?(0,4))
-    @x = x
-    @y = y
-    @facing = facing
-    @movements.replace(["PLACE #{@x}, #{@y}, #{@facing.to_s.upcase}"])
+    if valid_place(x,y)
```

Comment:
you won't need an else condition if you just raise the exception first.
``` if !valid(x,y) raise InvalidPositionError....````

Less indenting makes the code easier to read.